### ISSUE TITLE
Chat window steals focus froma game when it pops-up

### ISSUE THREAD

**Issue by Qik000** [ValveSoftware/steam-for-linux#5286]
#### Your system information
Arch Linux x64
GTX980 Properitary driver
Gnome 3.26

* Steam client version (build number or date): 16-12-2017
* Distribution (e.g. Ubuntu): Arch
* Opted into Steam client beta?: [Yes/No] No
* Have you checked for system updates?: [Yes/No] Yes

#### Please describe your issue in as much detail as possible:
Describe what you _expected_ should happen and what _did_ happen. Please link any large code pastes as a [Github Gist](https://gist.github.com/)
Chat window steals window focus when somwone sends a message, but leaving it open in background fixes the issue.

#### Steps for reproducing this issue:

1. Open any game
2. Make someone send you a message
3. Observe keybaord input and mouse input taken away from a game into the chat window

**Comment by kisak-valve** [comment]
Hello @Qik000, do you happen to have the steam overlay disabled with the games you are seeing this issue with?

**Comment by Qik000** [comment]
I have overlay enabled. When i open the chat in the overlay it also opens a new separate chat window which steals the focus again.

**Comment by Qik000** [comment]
Do you know how to fix it @kisak-valve ?

**Comment by steamlinux** [comment]
same problem, game minimize when steam create chat window and gift too
debian sid 64 bit, gnome 3.26, nvidia 384.98, steam beta

**Comment by Qik000** [comment]
@steamlinux i think i doesnt matter that you opted into beta. It happens on every update channel

**Comment by Qik000** [comment]
@triage-valve can we get some help ? Or at least get info on how to disable chat window popups ?

**Comment by steamlinux** [comment]
try at 1st open chat with someone, before you start game

**Comment by Qik000** [comment]
@steamlinux it helps but its not exactly a solution. Valve could easily fix this.

**Comment by tsuehpsyde** [comment]
I have also run into this issue on my system:

Fedora 27 64-bit
Gnome 3.26
GTX 970 w/ proprietary driver

**Comment by dearlygparted** [comment]
Imagine trying to play an online multiplayer game, you get a message from a Steam friend and it causes the game to minimize. You quickly try to alt tab back into the game but you've died. GG. Fix this please, ASAP.

**Comment by Qik000** [comment]
@dearlygparted thats exactly wat happend to me. And since Ubuntu is now using Gnome Valve should really fix it ASAP.

**Comment by Qik000** [comment]
@triage-valve is anyone working on this issue ?

**Comment by admshao** [comment]
It happens EVERY DAY. Its always nice when it happens on the middle of a Team Fight while playing dota

**Comment by oblitum** [comment]
It's worse on i3. Let's say I leave Steam BigPicture open on a workspace, and I'm with the terminal emulator Termite in another workspace. If I'm on the workspace with the Termite taking the entire screen and someone from Steam chat communicates with me, a normal Steam chat window opens up and this happens:

- https://github.com/i3/i3/issues/2401#issuecomment-367519037

**Comment by hickop** [comment]
Settings > Friends > Flash window when I receive a message > Never
Even with this it pops up chat windows on Gnome 3.26
Worked fine in Xfce 4

**Comment by Qik000** [comment]
@hickop do you know any way how to fix this ? It seems that valve doesnt give a cup boout it.

**Comment by Kiibakun** [comment]
Valve pls, this must be fixed asap

**Comment by kisak-valve** [comment]
Looks like this is the the default behavior for any newly spawned windows and Gnome.

Check if running `gsettings set org.gnome.desktop.wm.preferences focus-new-windows 'strict'` in a terminal makes Gnome behave as expected.

Note: This setting can be reverted to default with `gsettings set org.gnome.desktop.wm.preferences focus-new-windows 'smart'`.

**Comment by oblitum** [comment]
I have to confess that after moving to i3 window manager all my problems are gone :)

**Comment by robobenklein** [comment]
I have the new focus setting set to `strict` but the windows still steal focus anyways.

**Comment by Qik000** [comment]
@kisak-valve that works on everything EXCEPT steam...

**Comment by vibhas77** [comment]
Tried using all the above mentioned but still no solution yet.

**Comment by Qik000** [comment]
@kisak-valve 
I switched to Ubuntu 18.04 which is an LTS build, and you claim that steam is working perfectly on LTS, well ITS NOT. With the stock Gnome from 18.04 EXACT same thing happens. Can you finally fix it ?

**Comment by Vash63** [comment]
I'm also having this problem on Arch with Gnome 3.28 on both the stable and beta chat clients.

**Comment by furai** [comment]
Yup, this issue is annoying to say the least. In old chat I just used to keep chat window open - that wouldn't minimize the game while playing and receiving a message. But with new chat beta it's worse - it minimizes the games on every single message received. No matter if window was already spawned or not.

**Comment by Qik000** [comment]
@triage-valve  can you filanny do something about you program on a system that you claim that steam WORKS PERFECTLY ? It doesnt work as intended, setting windows policy to strict affects all applications EXCEPT steam !

CAN YOU FINALLY FIX THIS AFTER ALL THIS TIME ?
And also the notofication popup in the corner laggs the whole Gnome while its moving.
Is there anything else you will leave broken ?!

**Comment by robobenklein** [comment]
@mdesantis I wouldn't really consider that a workaround, since the keyboard input is still shifted from the SDL application to the new Steam window, the envvar doesn't actually prevent the change of window focus at all.

**Comment by mdesantis** [comment]
@robobenklein you're right, it doesn't even work.

**Comment by timjstacey** [comment]
I have the same issue on Pop!_OS using Gnome 3.28.2.  I don't have this issue on KDE.

**Comment by headlesscyborg** [comment]
@qik000 I'm 100% sure they never said that.

I am however experiencing the same on Arch with Gnome 3.28. I got used to it and learned to open a chat window before playing anything (Steam beta does it automatically now).

**Comment by Qik000** [comment]
@kisak-valve still NOT FIXED in the new update.

**Comment by mdesantis** [comment]
And it's even more annoying now, since with new Friends UI is not possible to disable friends at Steam startup!

**Comment by timjstacey** [comment]
I stumbled over something here that seems to do the trick for me https://superuser.com/questions/143044/how-to-prevent-new-windows-from-stealing-focus-in-gnome

`sudo apt install compizconfig-settings-manager`
`ccsm`

Go to General > General Options > Focus & Raise Behaviour.
Change Focus Prevention Level to normal

Not sure what impact this has in Gnome, or if it's actually fixing the problem.  Seems to be working so far.

**Comment by steamlinux** [comment]
maybe gnome have similar option on dconf?

**Comment by steamlinux** [comment]
I type "focus" on dconf and see some options. maybe someone know, what need to change? idk

**Comment by kisak-valve** [comment]
Hello @steamlinux, see https://github.com/ValveSoftware/steam-for-linux/issues/5286#issuecomment-390438395 and discussion following it.

**Comment by steamlinux** [comment]
new behavior? hm....
but, if other app create new window, I doesn't focus to him
only steam always get focus. example while run and install updates (without -silent option). steam, while start, create some windows (install update, connect to profile, open library) and always steam windows get focus. another apps doesn't do this

**Comment by robobenklein** [comment]
@timjstacey What DE does that work for you on? Gnome Shell is using metacity, not compiz, so my first thought would be that it wouldn't have the same effect.

**Comment by timjstacey** [comment]
@robobenklein I'm using Gnome 3.28.2 and you are right, it didn't fix it.  I have a friend in steam that when ever I opened a chat with them in game, the desktop window would steal focus and dump me onto desktop.  After running compiz I opened a chat window with that same friend and didn't get dumped to desktop, so thought it had fixed the issue.  After testing a little further, I realised that the opening a chat window in steam overlay with the last person you spoke with will dump you onto desktop.

To replicate:
Start a game, in my case cs:go
Open steam overlay
Open friends list
Open chat window with the last friend that sent you a message in your previous session

Result:
Desktop chat window steals focus

The only successful workaround I have found is to open a chat window with the last person you spoke with before starting a game.  Hardly the best work around, but seems to stop the issue from reoccurring in the testing I have done.

**Comment by Qik000** [comment]
@timjstacey its not really a workaround. Ots a hassle

**Comment by Qik000** [comment]
@headlesscyborg and they said to me a while ago back when i was on arch and i wrote on the steam support: "We can only guarantee steam`s functionality on Ubuntu [...] we cannot help you if you are not using the Ubuntu OS"

**Comment by ovollbrecht** [comment]
I came here from Windows and its the damn same desaster! It's driving me insane..

**Comment by steamlinux** [comment]
I report issue about sound on l4d2 in 2014, valve doesn't fix it (4 year)
how many time for this issue? hmmm....

**Comment by AlexsanderAR** [comment]
Same issue here. Every single chat message steals the focus minimizing the game.
It's so annoying and frustrating.
Now I'm playing with chat offline to prevent that.
Ubuntu 18.04
Gnome 3.28
Game: Dota2

**Comment by lucifertdark** [comment]
For me the chat window is stealing focus from the entire system every time it pops up, is there any way to stop it popping up at all unless I want to use it?
Ubuntu 18.04
Gnome 3.28

**Comment by vmavromatis** [comment]
I hate to spam threads, but can you please fix this issue as it's been almost a year? Thank you

**Comment by dearlygparted** [comment]
Hello, can we get an update on this? 

This is a real problem especially given other issues like this one:

https://github.com/ValveSoftware/Dota-2/issues/985

**Comment by steamlinux** [comment]
one solution at this moment - remember for open chat, before start the game

**Comment by ghanendrapiplani** [comment]
i am having the same issue on ubuntu 18.04, cannot believe this is being faced by all the linux users still valve wont fix it!! :(

**Comment by Strykar** [comment]
Same issue on Ubuntu 18.10, Dota 2 alt-tabs out to the message window when someone PMs me.
Occasionally, when Steam starts, it will load previous chat messages and some of the tabs appear as unread, when in fact, I have read and closed those chat messages before.

**Comment by kaadmy** [comment]
Another issue I'm having of a similar vein is that Steam gets forced focus sometimes (mainly when starting a game), and switches to the workspace that Steam is running in. Using i3wm.

**Comment by steamlinux** [comment]
for me steam always get focus? when create some window

**Comment by dauddimas** [comment]
[Complaint Part]
Same thing happens to me. I have to change my username in steam "Don't whisper me while playing". And even if my friends knew the issue I have, sometimes they forget and still whisper me, which leads to get blamed by my teammates and lose the game. Although I'm only playing dota2 but I have pretty much MMR, that's why it's very unpleasant since my games now getting intense and harder, every second counts. Hope valve will fix it soon.

[Question Part]
Does KDE have the same issue? Installed KDE not long ago because of curiosity, thanks for your answer.

**Comment by OvermindDL1** [comment]
> Does KDE have the same issue? Installed KDE not long ago because of curiosity, thanks for your answer.

At least for me I've not had the chat window grab focus, I've had multiple times where it's been blinking away in the background trying to get my attention but I don't see it because the game is focused.  Might be a setting I use though?

**Comment by OvermindDL1** [comment]
> Any settings you've changed? Wanna get workaround for this problem, but pretty confused of windows setting in KDE Tweak, since I'm a newcomer. I have no more patience, because even accidentally press menu hotkey would steal focus of the game ._.

Probably, but I've been running this install for so long that I'm not sure what is 'standard' about it anymore...  ^.^;

What do you mean by 'menu hotkey'?  Steam's overlay?  Something else?

**Comment by headlesscyborg** [comment]
Th issue doesn't bother my anymore since the new Steam chat update a few months ago - Steam now opens a chat window automatically when I launch it so it's not stealing focus anymore. Do you people manually close all chat windows before closing Steam itself?

**Comment by steamlinux** [comment]
I always use '-silent' option and steam ingnore tyis at last week
they stealing focus on startup, but message dialog doesn't create

**Comment by rastating** [comment]
Also experiencing this on Ubuntu 18.04

**Comment by rastating** [comment]
I've found a work around to this tonight. In the friends window, to the left of the minimise button will be a button to "Toggle chat window docking". If you click this, it will open new chat windows as a tab within the friends window. As long as you have the friends window open before launching the game, any new chats will become tabs of that window, and thus will stop your game minimising and focusing on a new window - as there isn't a new window being created.

Not the best of solutions, if you like to have separate windows for every open conversation, but it's certainly better than getting minimised out of a game at a crucial moment!

**Comment by Qik000** [comment]
> I've found a work around to this tonight. In the friends window, to the left of the minimise button will be a button to "Toggle chat window docking". If you click this, it will open new chat windows as a tab within the friends window. As long as you have the friends window open before launching the game, any new chats will become tabs of that window, and thus will stop your game minimising and focusing on a new window - as there isn't a new window being created.
> 
> Not the best of solutions, if you like to have separate windows for every open conversation, but it's certainly better than getting minimised out of a game at a crucial moment!

But it still SHOULD`T be an issue, and be fixed.

**Comment by rastating** [comment]
> But it still SHOULD`T be an issue, and should be fixed.

Of course - I didn't suggest otherwise. Until such time as Valve fix it though, the above is a way to work around it; as opposed to not using Steam.

**Comment by Saroumane** [comment]
18 months later, bug still not fixed.
Thanks @rastating  for the workaround.

**Comment by soprendi** [comment]
Why is this still an issue? This is depressing!

**Comment by KatharineKatharsis** [comment]
it is appalling that valve has done _*checks notes*_ ZERO to fix this shit...

**Comment by Qik000** [comment]
@KatharineKatharsis look when i made this issue...

**Comment by dauddimas** [comment]
> I've found a work around to this tonight. In the friends window, to the left of the minimise button will be a button to "Toggle chat window docking". If you click this, it will open new chat windows as a tab within the friends window. As long as you have the friends window open before launching the game, any new chats will become tabs of that window, and thus will stop your game minimising and focusing on a new window - as there isn't a new window being created.
> 
> Not the best of solutions, if you like to have separate windows for every open conversation, but it's certainly better than getting minimised out of a game at a crucial moment!

this solution works pretty well on high-tier computer, but for average computer like mine, this workaround costs your patience more than you think. since you have to be patient when all of a sudden your computer is freezing in-game, turns out it's your friend sends you message.

**Comment by KibaGasteiz** [comment]
> this solution works pretty well on high-tier computer, but for average computer like mine, this workaround costs your patience more than you think. since you have to be patient when all of a sudden your computer is freezing in-game, turns out it's your friend sends you message.

And what are your specs? Never had a problem with this.

**Comment by mscharstrom** [comment]
I'm also having issues with this. Still no solution to the issue?

**Comment by Qik000** [comment]
> I'm also having issues with this. Still no solution?

```
chmod 444 .steam/registry.vdf
chmod 544 .steam
```

**Comment by erikrecio** [comment]
Hey man, this issue just happened to me as well. I have Win10 and it never happened before, until i reinstalled Steam. I always had it in my main storage disc SD250, but now i put it in my secondary one with 2TB (not SD).

Probably this has something to do with the problem, at least for windows, although it shouldn't be happening. I'll use the chat solution for now, hope it gets fixed.

**Comment by kakolisgay** [comment]
+1 It has been almost two years and this bug still hasn't been resolved, could someone at least be assigned to this?

**Comment by KibaGasteiz** [comment]
This, i just have the merged window activated so any new friend message won't create a new window, but it's annoying.

**Comment by at94official** [comment]
Good Day, Valve Team. Still no solution for this bug? It's bothering.

**Comment by kakolisgay** [comment]
> This, i just have the merged window activated so any new friend message won't create a new window, but it's annoying.

I asked about this on the Steam LUG and one person suggested using do-not-disturb, this fixed the issue for me although now I don't get any chat notifications. Not a great solution but at least I don't get pulled out of my game every time someone pings me

**Comment by Xerxes-8933AA** [comment]
This happens on current Ubuntu / Pop!_OS 19.10, it's apparently gnome related which is the default window manager on these OS.

One work around is to enable "dock chats to the friends list" from "friends list" then 'settings -> chat'. This prevents new windows from opening and so wont steal focus and minimize your game.

This bug is particularly bad on Pop!_OS as the custom gnome front end they use has a top bar that causes a game malfunction once focus is stolen. Alt-tab to switch back to your game after receiving a chat message the full-screen view will be shifted down about 30 pixels from where the hit box for menus is located. I noticed this behaviour in Left for Dead 2, probably does it in other source engine games as well but I haven't tested. The bottom of the full screen view is cut-off, and the top has a bar that is now not redrawn. So you can't even use the game after alt-tabbing back.  A severe bug as it destroys an online game every time you get a message, not acceptable!

There surely must be a work around for this that can be implemented in the steam client. The end user should not be having this experience on the officially supported Linux distribution of steam !?

**Comment by Xerxes-8933AA** [comment]
It's totally fucked actually if you open a chat window at all (ie, to send a message, not just receiving) from within the steam overlay the source engine game is minimized and then broken when you tab back.  Retarded.

**Comment by echaskaris** [comment]
"should your issue not be resolved immediately, it does not mean that a resolution is not being researched or tested"

**Comment by Xerxes-8933AA** [comment]
Hey thanks guys, I realise you guys are probably trying to find a way to
make it work, but unless people complain to reinforce the issue is ongoing
and its severity I think some bugs get forgotten about. Overall it's really
impressive the support Linux now has with steam. Fantastic, really.

**Comment by rindo-eiji** [comment]
Having the same issue on manjaro-gnome.
whenever i play Dota I lose the focus of the window when someone messages me. and they do that everytime I play. in crucial moments.
I tried the dock chat window to the friends list. hope it works.
I know there is a lot of DE in the linux world but I really hope someone is working on this.

**Comment by takdw** [comment]
> Having the same issue on manjaro-gnome.
> whenever i play Dota I lose the focus of the window when someone messages me. and they do that everytime I play. in crucial moments.
> I tried the dock chat window to the friends list. hope it works.
> I know there is a lot of DE in the linux world but I really hope someone is working on this.

I went around this problem, by leaving one chat window open at all times. The focus steal happens only when its opening up a new chat window. So if you leave a chat window open, the next new chat will just be added to the same window but in a different tab, and there will be no focus loss from the game.

**Comment by terndrup** [comment]
This really needs to be fixed, its still around in February 2020

**Comment by dearlygparted** [comment]
> I went around this problem, by leaving one chat window open at all times. The focus steal happens only when its opening up a new chat window. So if you leave a chat window open, the next new chat will just be added to the same window but in a different tab, and there will be no focus loss from the game.

Yes, leaving a chat window open works (there is an option to have new chats open a tab rather than a new window which is enabled by default).

Another workaround is to enable Do Not Disturb mode (you'll get notifications for any received messages, but it will prevent new chat windows from popping up automatically).

**Comment by Xerxes-8933AA** [comment]
So I noticed recently an update fixed the issue with the top bar in gnome causing the full screen to be offset once you alt-tab back to the game.  At least it's not terminal anymore if the game minimizes due to a chat message, or if you alt-tab out of it to check other software.

So as of now, in gnome 3.34.2 at least on Pop!_OS 19.1, the alt-tab works with full screen.

If you have "dock chat windows to friends list" enabled then the chat messages no longer steal focus as long as you start the game with the friends list open. If you had the friends list window closed the game will minimize still on the first message, but then subsequent messages will not steal focus anymore if you leave it open. Alt-tab works to get back to the game. Great.

So that's partially fixed, but much better behaviour now.

**Comment by udekmp** [comment]
Hey I think this is the same problem I had on both Ubuntu and Pop!_OS 20.04 with GNOME.  The two workarounds above me work fine.

**Comment by box2-** [comment]
This is now happening on Windows too.

**Comment by Xerxes-8933AA** [comment]
Lol.

**Comment by robobenklein** [comment]
Got really tired of this issue so I wrote a small program to grab focus back when it's stolen: https://github.com/robobenklein/protectmyfocus

Has been working for me and prevents chat windows, etc. from stealing focus from games, browsers, terminals, etc. It should work for any window (hooks on X11 events) but having focus taken and immediately returned might mess with fullscreen apps.

I obviously don't have every game so I'd like some help testing.

**Comment by soprendi** [comment]
Good job

**Comment by guustflater** [comment]
Same problem here (Ubuntu 20.04)

**Comment by AFPV-au** [comment]
Further on this issue;

It worth nothing that having the "friends" list open actually has a negative impact on frame rates in competitive games (such as liftoff and counterstrike).. I notice a drop of around 50FPS if any embedded content is present in the text, when the chat is left open. On this quad core cpu when im otherwise getting 300fps. It's not enough to leave it open you need to close it sometimes after replying (yes I have turned off content embedding, I still get performance drops) so the issue returns when they send another message.. I've taken simply to logging out of friends now when I go to steam. Over it frankly.

> Got really tired of this issue so I wrote a small program to grab focus back when it's stolen: https://github.com/robobenklein/protectmyfocus

Nice work robobenklein, I'll give it a try.

I wonder why valve can't work that out and fix steam messenger, if this a a gnome issue, isn't it obvious by now they are NOT going to change the behaviour, so maybe its time to fix the steam client?

**Comment by jcnils** [comment]
Same problem on manjaro. Going offline and online again seems to fix the for a while

**Comment by CaptainNoBeard** [comment]
Problem still exists on Ubuntu 20.04.

**Comment by Moskonfyt** [comment]
I FIXED IT!
I have been a linux user for about 8 months now and used Manjaro to ease myself into Linux, there have been a few frustrations but the other day I had to draw the line and discovered Pop OS, everything was nice and dandy till I started playing Csgo. The fps was on par and felt good until the invites started, so I started searching and found no results, so I told myself I'll give it a month and see if I can resolve it.. so I did it in the first day. I started with the desktop customization and found it's very limited (actually a lot) but that could be because I'm a new Linux user and still need some experience, I got used to XFCE and wanted that back. 

The search began and quickly found the command to transform to XFCE and whuala you won't believe it my Csgo is fixed... can receive messages without me loosing my focus in a comp and invited can flow again... there is still one thing that worked on Manjaro but not in Pop Os tho and it's Windowed Full Screen, because I have 3 monitors with Manjaro the game won't minimize when working on another screen during a game/overwatch... I find it strange that Pop Os haven't been paying attention to the community seeing as there is a full team working to polish the system? I hope my solution help you guys, and really hope someone from Pop Os fix the windowed Full screen otherwise I'll keep on tinkering till it's fixed or...

**Comment by soprendi** [comment]
> I FIXED IT!
> I have been a linux user for about 8 months now and used Manjaro to ease myself into Linux, there have been a few frustrations but the other day I had to draw the line and discovered Pop OS, everything was nice and dandy till I started playing Csgo. The fps was on par and felt good until the invites started, so I started searching and found no results, so I told myself I'll give it a month and see if I can resolve it.. so I did it in the first day. I started with the desktop customization and found it's very limited (actually a lot) but that could be because I'm a new Linux user and still need some experience, I got used to XFCE and wanted that back.
> 
> The search began and quickly found the command to transform to XFCE and whuala you won't believe it my Csgo is fixed... can receive messages without me loosing my focus in a comp and invited can flow again... there is still one thing that worked on Manjaro but not in Pop Os tho and it's Windowed Full Screen, because I have 3 monitors with Manjaro the game won't minimize when working on another screen during a game/overwatch... I find it strange that Pop Os haven't been paying attention to the community seeing as there is a full team working to polish the system? I hope my solution help you guys, and really hope someone from Pop Os fix the windowed Full screen otherwise I'll keep on tinkering till it's fixed or...

Not all heroes wear capes!

**Comment by KibaGasteiz** [comment]
> I FIXED IT!

I don't understand, what we need to do to fix this?

**Comment by echaskaris** [comment]
> > I FIXED IT!
> 
> I don't understand, what we need to do to fix this?

From what I understand, switch to xfce desktop manager

**Comment by Chiitoo** [comment]
Several different window managers offer varying levels of 'focus stealing prevention', which may help at least some people with this kind of issues.  It's not a perfect solution, nor a fix, but perhaps better than nothing (I haven't tried how Steam would react with, say, 'extreme' level of prevention, but if it gets past that, I'd imagine it would take this issue to another level).

**Comment by Moskonfyt** [comment]
Just some feedback regarding Pop Os, after switching to Xfce desktop a few things happened regarding repositories and struggled to get that fixed... tried a few repositories but nothing worked, started the old fashion apt-get update in terminal after that had some authentication issues. Long story short I moved over to MX and boy is this a treat, everything works as it should Windowed Full screen everything and the speed is amazing... if anyone want to move over from ubuntu based distros give MX a go.

**Comment by Meister1593** [comment]
I have pretty nasty issue on AwesomeWM, when someone starts chatting to me in-game, whole screen starts being obstructed by steam chat because i use tiling by default, and game is full screen (not entirely on-top for everything) - so i have to alt-tab to chat and move it somewhere else.
Thought i think it's quite easy to fix for me - just make a rule for such thing, but i had somewhat similar issues in other DE's and wm's

**Comment by bluelightzero** [comment]
Seems like the simplest solution is not spawn the window in the first place.

The only reason a new window should ever pop-up is through user action.

**Comment by Rodancoci** [comment]
Any update on this? It's been years and so far all users have are undependable or cumbersome workarounds they've found themselves.

**Comment by bluelightzero** [comment]
@Rodancoci Seems setting your status to do not disturb stops the chat window spawning.

![image](https://user-images.githubusercontent.com/4492982/114785283-64552180-9d74-11eb-845e-b251955b0e27.png)

**Comment by Rodancoci** [comment]
Does this prevent chats from opening in the Steam Overview?

**Comment by Meister1593** [comment]
> @Rodancoci Seems setting your status to do not disturb stops the chat window spawning.
> 
> ![image](https://user-images.githubusercontent.com/4492982/114785283-64552180-9d74-11eb-845e-b251955b0e27.png)

Sure this can help, but i want notifications without chat window spawn. Not entire notifications removal

**Comment by chrisserung** [comment]
SDL_VIDEO_MINIMIZE_ON_FOCUS_LOSS=0 %command%
solved it for me on GNOME 3.38.4

**Comment by KibaGasteiz** [comment]
> SDL_VIDEO_MINIMIZE_ON_FOCUS_LOSS=0 %command%
> solved it for me on GNOME 3.38.4

Sadly, i don't see myself adding that launch option to every entry in my library of hundreds of games.

**Comment by Meister1593** [comment]
> > SDL_VIDEO_MINIMIZE_ON_FOCUS_LOSS=0 %command%
> > solved it for me on GNOME 3.38.4
> 
> Sadly, i don't see myself adding that launch option to every entry in my library of hundreds of games.

You can add this line in ~/.profile like so:
export SDL_VIDEO_MINIMIZE_ON_FOCUS_LOSS=0
This way every program will inherit this env var
Or you can pass that variable to steam in steam .desktop file, but this is more complicated

**Comment by AFPV-au** [comment]
> SDL_VIDEO_MINIMIZE_ON_FOCUS_LOSS=0 %command%
> solved it for me on GNOME 3.38.4

> You can add this line in ~/.profile like so:
> export SDL_VIDEO_MINIMIZE_ON_FOCUS_LOSS=0

Confirmed this works on gnome 3.36.8, you legends.

So to be specific it's not a perfect fix, but it's really close.  The first message you get will pop up on top of your game without stealing focus, and then first mouse input (or keyboard) such as movement will make it get overwritten.  So if you are moving it will disappear instantly, which is a massive improvement as you will not get locked up, just see a flash.  

Hopefully this doesn't cause glitches in other software..

**Comment by AFPV-au** [comment]
> You can add this line in ~/.profile like so:
> export SDL_VIDEO_MINIMIZE_ON_FOCUS_LOSS=0
> This way every program will inherit this env var on next login
> Or you can pass that variable to steam in steam .desktop file, but this is more complicated

Well I tried this.  It does seem to prevent the game from minimising, and being replaced with the chat window.  However it now pops up in front of the game, and then due to the mouse being in front of it (focus follows mouse is on)  takes focus anyway requiring you to click off it.  Its a minor improvement but still retarded and broken behavior, it will still definitely get you killed or loose a game when you receive a message.

**Comment by Meister1593** [comment]
> > You can add this line in ~/.profile like so:
> > export SDL_VIDEO_MINIMIZE_ON_FOCUS_LOSS=0
> > This way every program will inherit this env var on next login
> > Or you can pass that variable to steam in steam .desktop file, but this is more complicated
> 
> Well I tried this. It does seem to prevent the game from minimising, and being replaced with the chat window. However it now pops up in front of the game, and then due to the mouse being in front of it (focus follows mouse is on) takes focus anyway requiring you to click off it. Its a minor improvement but still retarded and broken behavior, it will still definitely get you killed or loose a game when you receive a message.

Yeah, unfortunately...:
1. this works only in SDL-based games (source engine games)
2. this prevents game from minimizng on focus loss literally, so if you have multiple screens - it won't minimize when you will focus on another app on another screen. This does not prevent from apps appearing in front of game

In the end this does not help at all with this issue

**Comment by emi-zqrd** [comment]
Same problem here with Ubuntu  20.04 LTS

**Comment by vibeisveryo** [comment]
still happens

**Comment by HannesMann** [comment]
Same issue here (GNOME 3.38 on Arch). Shouldn't the chat window open in the overlay?

**Comment by yonderbread** [comment]
Still broken, most annoying thing ever.

**Comment by yonderbread** [comment]
> how is this still a thing in 2021???

No idea, the fact that it does mildly infuriates me.

**Comment by Sarthak001** [comment]
any updates on this issue ??

**Comment by yonderbread** [comment]
> any updates on this issue ??

Unfortunately not as far as I am aware.

**Comment by JesseBusman** [comment]
This is still an issue. Please add an option to not have any new windows spawn automatically.

**Comment by jr1991-g** [comment]
Steam Deck was released, and this keeps happening. This bug it's almost five years old. 

I wonder if this will end up being Valve's "thumbnails in the file picker". I'm sorry to clutter this issue tracker with despair.

**Comment by barfin** [comment]
this has been a known bug for MANY YEARS 
valve is worth more than 7 billion and yet they can't fix the bug that affects millions of their users
this is unbelievable

**Comment by Sarthak001** [comment]
i think this issue got fixed !!! my game no longer getting minimize ( csog and dota 2 ) 
OS : LINUX ( 5.15.57-2-MANJARO )
steam ( opted for beta )

**Comment by jr1991-g** [comment]
Seems fixed on latest steam beta

**Comment by kuraida** [comment]
Still seems to be an issue with latest beta
Steam Flatpak (opted in for beta)
Ultramarine Linux 36

**Comment by reDBo0n** [comment]
Can confirm its still an issue in the beta.
Steambuild: 1683078372
Arch Linux

**Comment by MateusRodCosta** [comment]
From the June 16th Beta:

![Captura de tela de 2023-06-18 16-01-55](https://github.com/ValveSoftware/steam-for-linux/assets/24463334/77541e28-aec6-4543-8358-8c39b0001be3)

You guys might want to test.

I wonder if it also fixes #3783, which seems to be related (this one for Chat, the other one for Steam proper).

**Comment by jr1991-g** [comment]
This seems to be fixed. I didn't experience it in months since [this](https://github.com/chromium/chromium/commit/3db611c6ea0da69fba09ff2f8abe12d216f77e41), nor in Ubuntu 22.04 or Fedora 38. No idea about flatpak or snaps, just regular deb and rpm installers.
Sadly, #3783 it's still there.

[Steam info](https://gist.github.com/jr1991-g/6e7e4d57d714feef80a5af2515f01be9)

**Comment by epssy** [comment]
This issue persists with the latest Steam beta using Ubuntu 23.04 running X11.  When the chat window is closed any new message will minimise Dota 2, usually in the middle of a team fight.

**Comment by reDBo0n** [comment]
Since the new In-Game Overlay Update I did not experience this problem at all

**Comment by taisph** [comment]
I haven't had this issue since the big UI update.

**Comment by crystaldaking** [comment]
The problem still exists 
Ubuntu 23.10, X11, Steam (deb/snap)
